logstash/class-logger.php

```diff
@@ -535,4 +540,28 @@ public static function wp_debug_log( array $entry ) : void {
 			);
 		}
 	}
+
+	/**
+	 * Get maximum log entry count, any additional log entry after that will be silently discarded.
+	 *
+	 * @return int max number of entries
+	 */
+	public static function get_log_entry_limit() {
+		return defined( 'WP_CLI' ) && WP_CLI ? self::MAX_ENTRIES * 5 : self::MAX_ENTRIES;
+	}
+
+	/**
+	 * Get user agent for request if available
+	 *
+	 * @return string
+	 */
+	public static function get_user_agent() {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return 'cli';
+		} else {
+			// Match handling in other places
+			// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+			return substr( sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ?? '' ), 0, 255 );
+		}
+	}
```

Comment:
Small nitpick, but removes the `else` :)

```suggestion
	public static function get_user_agent() {
		if ( defined( 'WP_CLI' ) && WP_CLI ) {
			return 'cli';
		}
		
		// Match handling in other places
		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
		return substr( sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ?? '' ), 0, 255 );
	}
```